docs/index.html

```diff
@@ -96,49 +96,62 @@ <h1>Marked.js Documentation</h1>
           window.fetch = unfetch;
         }
 
-        if (!Element.prototype.closest) {
-
-            if (!Element.prototype.matches) {
-                Element.prototype.matches = Element.prototype.msMatchesSelector ||
-                                            Element.prototype.webkitMatchesSelector;
-            }
-
-            Element.prototype.closest = function(s) {
-                var el = this;
-                if (!document.documentElement.contains(el)) return null;
-                do {
-                    if (el.matches(s)) return el;
-                    el = el.parentElement || el.parentNode;
-                } while (el !== null && el.nodeType === 1);
-                return null;
-            };
-        }
-
         var content = document.querySelector('#content');
         var body = document.querySelector('html');
+        var currentPage = 'README.md';
+        var currentHash = '';
+        var renderedPage = '';
+
+        function hashChange() {
+            var hash = location.hash.slice(1);
+            if (!hash) {
+                hash = 'README.md';
+            }
+
+            var uri = hash.split('#');
 
-        body.addEventListener('click', function (e) {
-            var a = e.target.closest('a');
-            if (a && a.href.indexOf(location.origin) === 0) {
-                var uri = a.href.slice(location.origin.length + location.pathname.length).split('#');
-
-                if (uri[0].match('.md$')) {
-                    e.preventDefault();
-                    fetchPage(uri[0])
-                        .then(function () {
-                            if (uri.length > 1) {
-                                location.hash = uri[1];
-                            }
-                        });
+            if (uri[0].match('.md$')) {
+                currentPage = uri[0];
+                if (uri.length > 1) {
+                    currentHash = uri[1];
+                } else {
+                    currentHash = '';
                 }
-                history.replaceState("", document.title, "/");
+            } else {
+                currentHash = uri[0];
             }
-        }, false);
+
+            fetchPage(currentPage)
+                .then(function () {
+                    if(currentHash) {
+                        var hashElement = document.getElementById(currentHash);
+                        if (hashElement) {
+                            hashElement.scrollIntoView();
```

Comment:
Change `hashElement.scrollIntoView();` to `hashElement.scrollIntoView({ behavior: 'smooth' });` for a buttery transition ⛷️

Reply:
It doesn't seem to work very well.

![screenshot](https://user-images.githubusercontent.com/97994/37985640-f1c2ca10-31be-11e8-8a42-5e20f55884f6.gif)

Reply:
It worked correctly when you clicked "supported specifications" because that was the only time it hit the `scrollIntoView()` function as far as I can tell.